lib/charms/postgresql_k8s/v1/postgresql.py

```diff
@@ -1058,6 +1060,11 @@ def set_up_database(self, temp_location: Optional[str] = None) -> None:
             cursor = connection.cursor()
 
             if temp_location is not None:
+                # Fix permissions on the temporary tablespace location when a reboot happens and tmpfs is being used.
+                user = pwd.getpwnam("_daemon_")
+                os.chown(temp_location, uid=user.pw_uid, gid=user.pw_gid)
```

Comment:
Should _change_owner() be moved to lib and used to avoid duplicates?

Reply:
I believe it's a good idea. I'd change it a bit and create a different Python file (and also move it to the single kernel library repo) to contain utilities related to the filesystem, and keep this library containing only things related to the database connection and interaction.